Bottom: 18c1ff3d9652065996b248372b40cf4cdde29926
Top:    77558717525762df3805ba1708185d90264f6e89
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-01-04 21:16:24 +0000

Refresh of demo-of-exporter-specific-expo

---

diff --git a/src/hid/common/draw_helpers.c b/src/hid/common/draw_helpers.c
index 6c17c95..d514b5b 100644
--- a/src/hid/common/draw_helpers.c
+++ b/src/hid/common/draw_helpers.c
@@ -768,30 +768,12 @@ hole_callback (const BoxType * b, void *cl)
   return 1;
 }
 
-static int
-pin_callback (const BoxType * b, void *cl)
-{
-  //DrawPlainPin ((PinType *) b, false);
-  return 1;
-}
-
-static int
-pad_callback (const BoxType * b, void *cl)
-{
-  //PadType *pad = (PadType *) b;
-  //if (FRONT (pad))
-    //DrawPad (pad, 0);
-  return 1;
-}
-
 void
 common_export_region (HID *hid, BoxType *region)
 {
   int plated;
   int nplated;
   int nunplated;
-  int top_group;
-  int bottom_group;
   int group;
   int save_swap = SWAP_IDENT;
   bool paste_empty;
@@ -799,28 +781,10 @@ common_export_region (HID *hid, BoxType *region)
   PCB->Data->SILKLAYER.Color = PCB->ElementColor;
   PCB->Data->BACKSILKLAYER.Color = PCB->InvisibleObjectsColor;
 
-  top_group    = GetLayerGroupNumberBySide (TOP_SIDE);
-  bottom_group = GetLayerGroupNumberBySide (BOTTOM_SIDE);
-
   /* draw all copper layer groups in group order */
   for (group = 0; group < max_copper_layer; group++)
-    {
-      if (gui->set_layer (0, group, 0))
-        {
-          if (DrawLayerGroup (group, region))
-            {
-              r_search (PCB->Data->via_tree, region, NULL, pin_callback, NULL);
-              r_search (PCB->Data->pin_tree, region, NULL, pin_callback, NULL);
-
-              if (group == top_group || group == bottom_group)
-                {
-                  SWAP_IDENT = (group == bottom_group);
-                  r_search (PCB->Data->pad_tree, region, NULL, pad_callback, NULL);
-                  SWAP_IDENT = save_swap;
-                }
-            }
-        }
-    }
+    if (gui->set_layer (0, group, 0))
+      DrawLayerGroup (group, region);
 
   count_holes (region, &nplated, &nunplated);
 
diff --git a/src/hid/gerber/gerber.c b/src/hid/gerber/gerber.c
index 19098e8..0912221 100644
--- a/src/hid/gerber/gerber.c
+++ b/src/hid/gerber/gerber.c
@@ -628,9 +628,9 @@ gerber_expose (HID * hid, BoxType *drawn_area, void *item)
   hidGC savepm = Output.pmGC;
 
   gui = hid;
-  Output.fgGC = hid_draw_make_gc ();
-  Output.bgGC = hid_draw_make_gc ();
-  Output.pmGC = hid_draw_make_gc ();
+  Output.fgGC = hid_draw_make_gc (&gerber_graphics);
+  Output.bgGC = hid_draw_make_gc (&gerber_graphics);
+  Output.pmGC = hid_draw_make_gc (&gerber_graphics);
 
   hid->graphics->set_color (Output.pmGC, "erase");
   hid->graphics->set_color (Output.bgGC, "drill");
diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index e20555a..3422a86 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -887,12 +887,12 @@ ghid_expose (const BoxType *drawn_area)
   HID *old_gui = gui;
 
   gui = &ghid_hid;
-  Output.fgGC = gui->graphics->make_gc ();
-  Output.bgGC = gui->graphics->make_gc ();
-  Output.pmGC = gui->graphics->make_gc ();
+  Output.fgGC = hid_draw_make_gc (&ghid_graphics);
+  Output.bgGC = hid_draw_make_gc (&ghid_graphics);
+  Output.pmGC = hid_draw_make_gc (&ghid_graphics);
 
-  gui->graphics->set_color (Output.pmGC, "erase");
-  gui->graphics->set_color (Output.bgGC, "drill");
+  hid_draw_set_color (Output.pmGC, "erase");
+  hid_draw_set_color (Output.bgGC, "drill");
 
   PCB->Data->SILKLAYER.Color = PCB->ElementColor;
   PCB->Data->BACKSILKLAYER.Color = PCB->InvisibleObjectsColor;
@@ -998,9 +998,9 @@ ghid_expose (const BoxType *drawn_area)
       gui->end_layer ();
     }
 
-  gui->graphics->destroy_gc (Output.fgGC);
-  gui->graphics->destroy_gc (Output.bgGC);
-  gui->graphics->destroy_gc (Output.pmGC);
+  hid_draw_destroy_gc (Output.fgGC);
+  hid_draw_destroy_gc (Output.bgGC);
+  hid_draw_destroy_gc (Output.pmGC);
   gui = old_gui;
 }
 
diff --git a/src/hid/ps/ps.c b/src/hid/ps/ps.c
index e2cc2b8..a457f0c 100644
--- a/src/hid/ps/ps.c
+++ b/src/hid/ps/ps.c
@@ -581,12 +581,12 @@ ps_expose (void)
   bool paste_empty;
 
   gui = &ps_hid;
-  Output.fgGC = gui->graphics->make_gc ();
-  Output.bgGC = gui->graphics->make_gc ();
-  Output.pmGC = gui->graphics->make_gc ();
+  Output.fgGC = hid_draw_make_gc (&ps_graphics);
+  Output.bgGC = hid_draw_make_gc (&ps_graphics);
+  Output.pmGC = hid_draw_make_gc (&ps_graphics);
 
-  gui->graphics->set_color (Output.pmGC, "erase");
-  gui->graphics->set_color (Output.bgGC, "drill");
+  hid_draw_set_color (Output.pmGC, "erase");
+  hid_draw_set_color (Output.bgGC, "drill");
 
   PCB->Data->SILKLAYER.Color = PCB->ElementColor;
   PCB->Data->BACKSILKLAYER.Color = PCB->InvisibleObjectsColor;
@@ -638,9 +638,9 @@ ps_expose (void)
   if (set_layer ("fab", SL (FAB, 0), 0))
     PrintFab (Output.fgGC);
 
-  gui->graphics->destroy_gc (Output.fgGC);
-  gui->graphics->destroy_gc (Output.bgGC);
-  gui->graphics->destroy_gc (Output.pmGC);
+  hid_draw_destroy_gc (Output.fgGC);
+  hid_draw_destroy_gc (Output.bgGC);
+  hid_draw_destroy_gc (Output.pmGC);
   gui = old_gui;
 }
